include/deal.II/grid/tria.h

```diff
@@ -1468,6 +1474,18 @@ class Triangulation : public Subscriptor
     distorted_cells;
   };
 
+  /**
+   * If add_periodicity() is called, this variable stores the given
+   * periodic face pairs on level 0 for later access during the
+   * identification of ghost cells for the multigrid hierarchy and for
+   * setting up the periodic_face_map.
+   */
+  std::vector<GridTools::PeriodicFacePair<cell_iterator> > periodic_face_pairs_level_0;
+
+  /**
+   * If add_periodicity() is called, this variable stores the active periodic face pairs.
+   */
+  std::map<std::pair<cell_iterator, unsigned int>, std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > > periodic_face_map;
```

Comment:
I've been thinking about the name of the data structure here - it looks a bit lengthy with all those pairs (and cell_iterator is a bit wasteful because each item stores a pointer to the triangulation - but it is handy). On the other hand, it is nice to have a clear statement of the involved data structures so I don't see a better solution. Any thoughts?